core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -166,6 +178,10 @@ public static Builder builder() {
   // set to null until we check if FILE_TREE is supported
   private Boolean fileTreeSupported = null;
   private FileHasher fileHasher;
+  private int timeout = -1;
+
+  private Timer timer = new Timer("DirectoryWatcherTimer");
```

Comment:
This can be final. Further set it in the constructor. As this way you can set it null if timeout is still <= 0.

Reply:
other final or constructor case, right ?